commit aed361adca9ff1d828843f3afc5d9140a2fa410d
Author: Bjorn Andersson <andersson@kernel.org>
Date:   Fri Aug 12 18:18:59 2016 -0700

    remoteproc: qcom: Introduce WCNSS peripheral image loader
    
    This introduces the peripheral image loader, for loading WCNSS firmware
    and boot the core on e.g. MSM8974. The firmware is verified and booted
    with the help of the Peripheral Authentication System (PAS) in
    TrustZone.
    
    Tested-by: John Stultz <john.stultz@linaro.org>
    Signed-off-by: Bjorn Andersson <bjorn.andersson@sonymobile.com>
    Signed-off-by: Bjorn Andersson <bjorn.andersson@linaro.org>

diff --git a/drivers/remoteproc/qcom_wcnss.h b/drivers/remoteproc/qcom_wcnss.h
new file mode 100644
index 000000000000..9dc4a9fe41e1
--- /dev/null
+++ b/drivers/remoteproc/qcom_wcnss.h
@@ -0,0 +1,22 @@
+#ifndef __QCOM_WNCSS_H__
+#define __QCOM_WNCSS_H__
+
+struct qcom_iris;
+struct qcom_wcnss;
+
+struct wcnss_vreg_info {
+	const char * const name;
+	int min_voltage;
+	int max_voltage;
+
+	int load_uA;
+
+	bool super_turbo;
+};
+
+int qcom_iris_enable(struct qcom_iris *iris);
+void qcom_iris_disable(struct qcom_iris *iris);
+
+void qcom_wcnss_assign_iris(struct qcom_wcnss *wcnss, struct qcom_iris *iris, bool use_48mhz_xo);
+
+#endif